x/sequencer/keeper/msg_server_create_sequencer.go

```diff
@@ -85,10 +83,25 @@ func (k msgServer) CreateSequencer(goCtx context.Context, msg *types.MsgCreateSe
 		RollappId:    msg.RollappId,
 		Metadata:     msg.Metadata,
 		Status:       types.Bonded,
-		Proposer:     !rollapp.Sealed,
 		Tokens:       bond,
 	}
 
+	// if this is the first sequencer, make it a PROPOSER
+	bondedSequencers := k.GetSequencersByRollappByStatus(ctx, msg.RollappId, types.Bonded)
+	isProposer := len(bondedSequencers) == 0
+	if isProposer {
+		k.SetProposer(ctx, sequencer.RollappId, sequencer.Address)
+	}
```

Comment:
cant you just check if proposer set already?

Reply:
yeah possible